Allow dots in the names, bug #95173.  Patch by Kerin Millar.
Re-sourced for 2.28.2

diff -Nur debianutils-2.28.2/run-parts.8 debianutils-2.28.2-no-bs-namespace/run-parts.8
--- debianutils-2.28.2/run-parts.8	2007-11-12 11:24:24.000000000 -0500
+++ debianutils-2.28.2-no-bs-namespace/run-parts.8	2007-12-21 13:13:48.000000000 -0500
@@ -27,7 +27,8 @@
 
 If neither the \-\-lsbsysinit option nor the \-\-regex option is given
 then the names must consist entirely of upper and lower case letters,
-digits, underscores, and hyphens.                           
+digits, underscores, hyphens, and periods. However, the name must not begin
+with a period.
                                                                                 
 If the \-\-lsbsysinit option is given, then the names must not end
 in .dpkg\-old  or .dpkg\-dist or .dpkg\-new or .dpkg\-tmp, and must
diff -Nur debianutils-2.28.2/run-parts.c debianutils-2.28.2-no-bs-namespace/run-parts.c
--- debianutils-2.28.2/run-parts.c	2007-11-12 11:24:24.000000000 -0500
+++ debianutils-2.28.2-no-bs-namespace/run-parts.c	2007-12-21 13:14:32.000000000 -0500
@@ -161,7 +161,7 @@
     return !regexec(&tradre, c, 0, NULL, 0);
   }
   else {
-    if (regcomp(&classicalre, "^[a-zA-Z0-9_-]+$", REG_EXTENDED | REG_NOSUB)) {
+    if (regcomp(&classicalre, "^[a-zA-Z0-9_-][a-zA-Z0-9._-]+$", REG_EXTENDED | REG_NOSUB)) {
       error("regex failure");
       exit(1);
     }
